server/aesdsocket.c

```diff
@@ -336,11 +491,13 @@ int main(int argc, char* argv[])
       syslog(LOG_INFO, "Daemon started successfully \r\n");
     }
   }
-
+  
   /**************************** START SOCKET SERVER ***************************/
+  
+  pthread_t time_thread;
+  pthread_create(&time_thread, NULL, logTimeStamp, NULL);                                          // Run time stamp in a separate thread //
```

Comment:
@Vignesh-Vadivel missing a `pthread_join` cleanup of this thread - this is what valgrind is telling you.